x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java

```diff
@@ -63,6 +74,88 @@ public Geometry toGeometry() {
         return shapeBuilder.buildGeometry();
     }
 
+    public Point firstPoint() {
+        return shapeBuilder.buildGeometry().visit(new GeometryVisitor<Point>() {
+            @Override
+            public Point visit(Circle circle) {
+                return new Point(circle.getLat(), circle.getLon(), circle.hasAlt() ? circle.getAlt() : Double.NaN);
+            }
+
+            @Override
+            public Point visit(GeometryCollection<?> collection) {
+                if (collection.size() > 0) {
+                    return collection.get(0).visit(this);
+                }
+                return null;
+            }
+
+            @Override
+            public Point visit(Line line) {
+                if (line.length() > 0) {
+                    return new Point(line.getLat(0), line.getLon(0), line.hasAlt() ? line.getAlt(0) :  Double.NaN);
+                }
+                return null;
+            }
+
+            @Override
+            public Point visit(LinearRing ring) {
+                return visit((Line) ring);
+            }
+
+            @Override
+            public Point visit(MultiLine multiLine) {
+                return visit((GeometryCollection<?>) multiLine);
+            }
+
+            @Override
+            public Point visit(MultiPoint multiPoint) {
+                return visit((GeometryCollection<?>) multiPoint);
+            }
+
+            @Override
+            public Point visit(MultiPolygon multiPolygon) {
+                return visit((GeometryCollection<?>) multiPolygon);
+            }
+
+            @Override
+            public Point visit(Point point) {
+                return point;
+            }
+
+            @Override
+            public Point visit(Polygon polygon) {
+                return visit(polygon.getPolygon());
+            }
+
+            @Override
+            public Point visit(Rectangle rectangle) {
+                return new Point(rectangle.getMinLat(), rectangle.getMinLon(), rectangle.getMinAlt());
+            }
+        });
+    }
+
+    public Double getX() {
+        Point firstPoint = firstPoint();
+        if (firstPoint != null) {
```

Comment:
All this logic could be re-written in a more compact way as `return firstPoint != null ? firstPoint.getLon() : null;`